beacon_node/http_api/src/lib.rs

```diff
@@ -2444,6 +2444,78 @@ pub fn serve<T: BeaconChainTypes>(
             },
         );
 
+    // POST validator/register_validator
+    let post_validator_register_validator = eth1_v1
+        .and(warp::path("validator"))
+        .and(warp::path("register_validator"))
+        .and(warp::path::end())
+        .and(chain_filter.clone())
+        .and(warp::addr::remote())
+        .and(log_filter.clone())
+        .and(warp::body::json())
+        .and_then(
+            |chain: Arc<BeaconChain<T>>,
+             client_addr: Option<SocketAddr>,
+             log: Logger,
+             register_val_data: Vec<SignedValidatorRegistrationData>| {
+                blocking_json_task(move || {
+                    let execution_layer = chain
+                        .execution_layer
+                        .as_ref()
+                        .ok_or(BeaconChainError::ExecutionLayerMissing)
+                        .map_err(warp_utils::reject::beacon_chain_error)?;
+                    let current_epoch = chain
+                        .epoch()
+                        .map_err(warp_utils::reject::beacon_chain_error)?;
```

Comment:
```suggestion
                    let current_epoch = chain
                        .slot_clock
                        .now_or_genesis()
                        .ok_or(BeaconChainError::UnableToReadSlot)
                        .map_err(warp_utils::reject::beacon_chain_error)?
                        .epoch(T::EthSpec::slots_per_epoch());
```

Otherwise will return an error when run before genesis. We can use `now_or_genesis` to assume the genesis slot before the genesis time.

I doubt it'll ever present as an actual problem, but sometimes we find these issues when waiting for testnets to start and get Discord questions about it.